### PR TITLE
Use TypeSet instead of TypeList for projects, floatingips and licenses attributes

### PR DESCRIPTION
**Pull request by ozerovandrei**
Use TypeSet for the following attributes:

- `servers`
- `quotas`
- `all_quotas`
- `resource_quotas`

For #47

### CODE REVIEW

**Comment by ozerovandrei** [comment]
@bill-rich this is ready for review.

Travis will fail on `make website-test` for now (because there is no `selvpc` provider in the Docker image).

File: selvpc/servers.go [hunk 1]
```diff
@@ -19,3 +25,11 @@ func serversMapsFromStructs(serverStructs []servers.Server) []map[string]interfa
 
 	return associatedServers
 }
+
+// hashServers is a hash function to use with the "servers" set.
+func hashServers(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["id"].(string)))
```
Comment (bill-rich) [inline selvpc/servers.go]: Why is bytes.Buffer used instead of just using fmt.Sprintf to generate a string?

Comment (ozerovandrei) [reply selvpc/servers.go]: Well, I just wanted to use same way in all `hashSomething` functions. But I agree, it's better to use `fmt.Sprintf` in cases where we don't check if attribute is set.
I will fix that in `hashServers` and `hashQuotas` functions.

**Comment by ozerovandrei** [comment]
@bill-rich I fixed latest issue.

**Pull request merged**